drake_ros_systems/include/drake_ros_systems/drake_ros.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef DRAKE_ROS_SYSTEMS__DRAKE_ROS_HPP_
+#define DRAKE_ROS_SYSTEMS__DRAKE_ROS_HPP_
+
+#include <rosidl_runtime_c/message_type_support_struct.h>
+#include <rclcpp/clock.hpp>
+#include <rclcpp/node_options.hpp>
+#include <rclcpp/qos.hpp>
+#include <rclcpp/serialized_message.hpp>
+#include <tf2_ros/transform_broadcaster.h>
+
+#include <functional>
+#include <memory>
+#include <string>
+
+#include "drake_ros_systems/drake_ros_interface.hpp"
+
+namespace drake_ros_systems
+{
+/// PIMPL forward declarations
+class DrakeRosPrivate;
+class Publisher;
+class Subscription;
+
+/// System that abstracts working with ROS
+class DrakeRos final : public DrakeRosInterface
```

Comment:
I'm curious as to your thoughts on class hiearchy, why create a high level class like `DrakeRos` to handle publishing, subscribing, clock, and transforms, rather than individual wrappers?

Reply:
I have to defer to @sloretz here, but I presume this was modeled after the `DrakeLcmInterface` type hierarchy. Arguably, it's not the exact same use case, but it gets us easy dependency injection for testing down the road.